### PR TITLE
Kubernetes presentation

### PR DESCRIPTION
**Pull request by AdrienLemaire**

### CODE REVIEW

File: 20180224_kubernetes/index.html [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    <div id="history_2" class="step slide" data-x="0" data-y="-1500">
+        <q>Then came the advent of <strong>Internet</strong></q>
+        <span class="notes">
+            With the advent of internet, we could remotely update software, and
+            devops were making their own scripts to automate parts of the process.
+        </span>
+    </div>
+
+    <div id="history_3" class="step slide text_align" data-x="1000" data-y="-1500">
+        <p>The next logical steps were PaaS and IaaS!</p>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:43]: PaaS and IaaS in one are very abrupt jump. Maybe split into two slides?

Comment (AdrienLemaire) [reply 20180224_kubernetes/index.html:43]: The talk is focused on Kubernetes, so I was worried to spend too much time in that first "history" part. Also, considering that today's most popular services are offering both PaaS and IaaS services, I was just gonna mention the difference noted in l.64~67.
If you still think I should split, I will :)

Comment (martinheidegger) [reply 20180224_kubernetes/index.html:43]: I think you should split because when I read it, I struggled hard to remember what PaaS and IaaS is ... don't use those abbr. regularly. If I struggle, others will too!

File: 20180224_kubernetes/index.html [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        <img src="static/history_3.jpg" />
+        <div class="notes">
+            Then came Platforms as a Software. Through APIs, not only could we update software but also the hardware running it.<br />
+            <ul>
+                <li>Red Hat OpenShift</li>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:48]: Historic + Popular order: OpenShift, Heroku, Google Cloud, Amazon AWS, Windows Azure, ...

File: 20180224_kubernetes/index.html [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    <p>For the best experience please use the latest <b>Chrome</b>, <b>Safari</b> or <b>Firefox</b> browser.</p>
+</div>
+
+<div id="impress">
+
+    <div id="history_1" class="step slide" data-x="-1000" data-y="-1500">
+        <q>A very long time ago... <br />Expert system administrators used cartridges, cassettes tapes, floppy disks and optical media</q>
+        <span class="notes">
+            Before the internet, people had to distribute software across cartridges,
+            then cassette tapes, then floppy disks, and optical media.
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html]: Succinct version: "Without internet: cartridges, cassette tapes, floppy disks & optical media"

File: 20180224_kubernetes/index.html [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        </span>
+    </div>
+
+    <div id="history_2" class="step slide" data-x="0" data-y="-1500">
+        <q>Then came the advent of <strong>Internet</strong></q>
+        <span class="notes">
+            With the advent of internet, we could remotely update software, and
+            devops were making their own scripts to automate parts of the process.
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:38]: Succinct version: "With Internet: remotely update software, automate everything (with scripts"

File: 20180224_kubernetes/index.html [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+        </span>
+    </div>
+
+    <div id="history_3" class="step slide text_align" data-x="1000" data-y="-1500">
+        <p>The next logical steps were PaaS and IaaS!</p>
+        <img src="static/history_3.jpg" />
+        <div class="notes">
+            Then came Platforms as a Software. Through APIs, not only could we update software but also the hardware running it.<br />
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:46]: Succinct version: "Level++: APIs to update software; Level+++: APISs to update hardware!"

File: 20180224_kubernetes/index.html [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+                <li>you name it...</li>
+            </ul>
+            And IaaS services
+            <ul>
+                <li>Most of the PaaS named previously also offer IaaS services</li>
+                <li>Google App Engine → Google Compute Engine</li>
+            </ul>
+
+            <span>
+                <strong>SaaS</strong>: deliver software to clients without them needing any installation (eg web apps)
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:62]: Succinct version: "Mind blow (ani gif): The server is the client! No installation!!! (web apps...)"

File: 20180224_kubernetes/index.html [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+            </span><br />
+            <span>
+                <strong>PaaS</strong>: Rely on 3rd parties to handle the infrastructure, and use their tools to launch instances, configure your network and install your software.
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:65]: Fancy version: "PaaS: Let others deal with infrastructure; My scripts will do the rest - Puppet master!"

File: 20180224_kubernetes/index.html [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+            </span><br />
+            <span>
+                <strong>IaaS</strong>: Automate infrastructure deployment through APIs, set rules to trigger actions.
+            </span>
+        </div>
+    </div>
+
+    <div id="history_4" class="step slide text_align" data-x="2000" data-y="-1500">
+        <q> IaC to the rescue</q>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:74]: Readable: "Script everything: Code to the rescue"

File: 20180224_kubernetes/index.html [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+            Infrastructure as Code IaC<br />
+            Ansible, Chef, Puppet, SaltStack, Fabric, Terraform …<br />
+            Process of managing and provisioning servers through definition files rather than configuration tools.<br />
+            Promoted for cloud computing, often confused with Infrastructure as a Service (IaaS)<br />
+            This brought forth 3 improvements: Cost reduction, faster execution speed, and restricted risk (remove errors and security violations).<br />
+        </div>
+    </div>
+
+    <div id="history_5" class="step slide" data-x="2000" data-y="-2300">
+        <q>Enter the world of containers</q>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:91]: Inception image needed 😆

File: 20180224_kubernetes/index.html [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+        <img src="static/history_5_docker.jpg" />
+        <div class="notes">
+            Docker is a technology providing containers (or OS-level virtualization).
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:94]: Humanreadable: "Docker lets OS run in fast sandboxes. So they can play well with other OS's."

File: 20180224_kubernetes/index.html [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+        </div>
+    </div>
+
+    <div id="history_6" class="step slide text_align" data-x="1000" data-y="-2300">
+        <p>A strong ecosystem was built around Docker</p>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:99]: Humanreadable: "People like docker and build lots of stuff with it."

File: 20180224_kubernetes/index.html [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+        <img src="static/history_6.png" />
+        <div class="notes">
+            Prometheus, Docker compose, CoreOS, Mesosphere, Flocker, Cloud 66
+            Helios, Zookeeper, Drone, Twistlock, Weave cloud, Kubernetes
+        </div>
+    </div>
+
+    <div id="history_7" class="step slide text_align" data-x="0" data-y="-2300">
+        <p>And we'll dive into today's most popular orchestration engine:</p>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:108]: Humanreadable: "Today we learn about the best way to work with docker"

File: 20180224_kubernetes/index.html [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+        <img src="static/history_7.png" />
+        <div class="notes">
+            <ul>
+                <li>Created by Google, heavily influenced by Google’s Borg system, announced in mid-2014</li>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:112]: Succinct: "Created by google" (note: it misses the target user group google was aiming at.

Note: Why is it important for the audience to know about Borg?

File: 20180224_kubernetes/index.html [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+                <li>Kubernetes v1 released on July 21, 2015. Google partnered with the Linux Foundation to form the Cloud Native Computing Foundation (CNCF)</li>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:113]: Succinct version (with popularity ordering):
- v1 released 2,5 years ago
- Used by: IBM, Oracle, Red Hat, CoreOS, Rancher Labs
- Now its the king 👸

File: 20180224_kubernetes/index.html [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+
+    <div id="title" class="step" data-x="0" data-y="0" data-scale="4">
+        <span class="try">What exactly is</span>
+        <h1>Kubernetes</h1>
+        <img src="static/title_interrogative.png" />
+    </div>
+
+    <div id="pods" class="step" data-x="850" data-y="3500" data-rotate="90" data-scale="5">
+        <p>
+        It has sets of <strong>containers</strong> like Docker called <strong>Pods</strong>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:129]: Consistent using of the word set/group helpful imo.

Succinct alternative: "1 _Pod_ = a set of containers and other things"

File: 20180224_kubernetes/index.html [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+        </p>
+        <img src="static/pods.svg" />
+        <div class="notes">
+            Group of containers that will be co-located on the same host machine and share the same resources and IP address.<br />
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:133]: Succinct version:
- **1** _Pod_ = **1** machine + **1** IP address.
- Pods can have: containers, hard-disks, network setup.

File: 20180224_kubernetes/index.html [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+            A pod can define a volume (local disk directory or network disk)<br />
+        </div>
+    </div>
+
+    <div id="nodes" class="step" data-x="3700" data-y="2800" data-rotate="180" data-scale="5">
+        <p>Pods run on <b>Nodes</b> <span class="thoughts">(worker machine)</span></p>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:139]: Succinct version: "**1** _Node_ ー **N** _Pods_"

File: 20180224_kubernetes/index.html [hunk 18]
```diff
@@ -0,0 +1,223 @@
...
+        <img src="static/nodes.svg" />
+        <div class="notes">
+            A node is a worker machine in Kubernetes and may be a VM or physical machine, depending on the cluster.<br />
+            You might have heard of master/minions before, and nodes were previously called minions.
+            Multiple Pods can run on one Node.<br />
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:144]: Succinct version: "Old name: minions"

File: 20180224_kubernetes/index.html [hunk 19]
```diff
@@ -0,0 +1,223 @@
...
+            Group of containers that will be co-located on the same host machine and share the same resources and IP address.<br />
+            A pod can define a volume (local disk directory or network disk)<br />
+        </div>
+    </div>
+
+    <div id="nodes" class="step" data-x="3700" data-y="2800" data-rotate="180" data-scale="5">
+        <p>Pods run on <b>Nodes</b> <span class="thoughts">(worker machine)</span></p>
+        <img src="static/nodes.svg" />
+        <div class="notes">
+            A node is a worker machine in Kubernetes and may be a VM or physical machine, depending on the cluster.<br />
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:142]: Succinct version:
- Node Type A: physical machine
- Node Type B: VM

File: 20180224_kubernetes/index.html [hunk 20]
```diff
@@ -0,0 +1,223 @@
...
+            You might have heard of master/minions before, and nodes were previously called minions.
+            Multiple Pods can run on one Node.<br />
+            Every Kubernetes Node runs at least:
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:145]: Succinct version:

_Node_ Vocabulary:
- _Kubelet_: Handles communication & manages _Pod_'s
- _Container Runtime:_ Docker, rkt to run apps

File: 20180224_kubernetes/index.html [hunk 21]
```diff
@@ -0,0 +1,223 @@
...
+            <ul>
+                <li>Kubelet, a process responsible for communication between the Kubernetes Master and the Node; it manages the Pods and the containers running on a machine.</li>
+                <li>A container runtime (like Docker, rkt) responsible for pulling the container image from a registry, unpacking the container, and running the application.</li>
+                <li>Containers should only be scheduled together in a single Pod if they are tightly coupled and need to share resources such as disk.</li>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:149]: "Container should..." ??? you titled this "every Kubernetes Nodes runs at least". Seems like the last list item belongs to a different thought process.

File: 20180224_kubernetes/index.html [hunk 22]
```diff
@@ -0,0 +1,223 @@
...
+            </ul>
+        </div>
+    </div>
+
+    <div id="kubelet" class="step" data-x="3600" data-y="1600" data-z="-15000" data-rotate="300" data-scale="0.2">
+        <h1>Kubelet</h1>
+        <q>An agent that runs on each node in the cluster. It makes sure that containers* are running and healthy in a pod</q><br />
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:156]: I would restructure this: You already introduce this as communcation & manager. How about explaining what is going on, i.e.:

- "_Kubelet_ + _Pod Spec_ → Running Pod"
- "_Kubelet_ restarts broken pods"
- "_Kubelet_ reports to _Master_ 🤫" _(Note: "master was not introduced before this section")_

File: 20180224_kubernetes/index.html [hunk 23]
```diff
@@ -0,0 +1,223 @@
...
+        <span class="footnote">* as described in Pod specification files</span>
+    </div>
+
+    <div id="master" class="step" data-x="4200" data-y="-2500" data-z="0" data-rotate="270" data-scale="5">
+        <p>These <b class="positioning">Minion</b> nodes are controlled by a <b class="rotating">Master</b> node, running master components*:</p>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:161]: Entertaining Version:

_Master_ tells _Nodes_ what to do. 🧠

Boring, unimportant details:

File: 20180224_kubernetes/index.html [hunk 24]
```diff
@@ -0,0 +1,223 @@
...
+        <div class="footnotes">
+            <b class="scaling">kube-apiserver</b>|<b class="scaling">etcd</b>|<b class="scaling">kube-scheduler</b>|<b class="scaling">kube-controller-manager</b>|<b class="scaling">cloud-controller-manager</b>
+            <p>* It's ok to ignore these components, GCP even abstracted them</p>
+        </div>
+    </div>
+
+    <div id="cluster" class="step" data-x="7500" data-y="-1000" data-scale="6">
+        <p>This set of <b>master</b> components and <b>worker nodes</b> make up the<br /> <b class="cluster">cluster</b></p>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:169]: Succinct version: **1** _Cluster_ = **1** _Master_ + **N** _Nodes_

File: 20180224_kubernetes/index.html [hunk 25]
```diff
@@ -0,0 +1,223 @@
...
+        <img src="static/cluster.svg" />
+    </div>
+
+    <div id="selectors" class="step" data-x="8300" data-y="2500" data-rotate="20" data-scale="4">
+        <p>Wait, there's more!</p>
+        <q>You can group pods, no matter on which node they are running, through <b>labels</b> and <b>selectors</b>!</q>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:175]: Succinct version: "You can group pods! Using **labels** and **selectors**!"

(Its clear that pods are on a machine, no need to mention that they are on different machines).

File: 20180224_kubernetes/index.html [hunk 26]
```diff
@@ -0,0 +1,223 @@
...
+    </div>
+
+    <div id="services" class="step" data-x="6000" data-y="4000" data-scale="2">
+        <p>Enter <b>Services</b> and <b>Controllers</b></p>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html]: Alternative version: "Two new Words: _Services_ and _Controllers_"

File: 20180224_kubernetes/index.html [hunk 27]
```diff
@@ -0,0 +1,223 @@
...
+    </div>
+
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html]: Am I missing something? What do Services and Controllers do???

File: 20180224_kubernetes/index.html [hunk 28]
```diff
@@ -0,0 +1,223 @@
...
+    <div id="summary" class="step" data-x="6200" data-y="5400" data-z="-100" data-rotate-x="-40" data-rotate-y="10" data-scale="2">
+        <q>Let's summarize what we've learnt</q>
+        <img src="static/summary.png" />
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:206]: The summary png doesn't highlight the vocabulary introduced (or seems to use other vocabulary).

File: 20180224_kubernetes/index.html [hunk 29]
```diff
@@ -0,0 +1,223 @@
...
+        <q>Here's another representation of a cluster</q>
+        <img src="static/summary-2.png" />
+    </div>
+
+    <div id="demo-1" class="step slide" data-x="-5000" data-y="1500"
+        data-scale="3" data-rotate-order="xyz" data-rotate-x="10"
+        data-rotate-y="60">
+        <!--data-scale="3" data-rotate-z="10" data-rotate-x="20">-->
+        <h1>Demo time</h1>
+        <p>Let's run a local pod using <b>docker-compose</b></p>
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html]: Docker-compose is an unknown term.

File: 20180224_kubernetes/index.html [hunk 30]
```diff
@@ -0,0 +1,223 @@
...
+</head>
+<body class="impress-not-supported">
+
+<div class="fallback-message">
+    <p>Your browser <b>doesn't support the features required</b> by impress.js, so you are presented with a simplified version of this presentation.</p>
+    <p>For the best experience please use the latest <b>Chrome</b>, <b>Safari</b> or <b>Firefox</b> browser.</p>
+</div>
+
+<div id="impress">
+
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:24]: There are slides missing here:

1. What are we going to learn about today and why is it important?
2. Who should know this?
3. Who are you, and why should we listen to **you**?

File: 20180224_kubernetes/index.html [hunk 31]
```diff
@@ -0,0 +1,223 @@
...
+            <img src="static/history_4_puppet.png" />
+            <img src="static/history_4_saltstack.png" />
+            <img src="static/history_4_fabric.png" />
+            <img src="static/history_4_terraform.png" />
+        <div class="notes">
+            Infrastructure as Code IaC<br />
+            Ansible, Chef, Puppet, SaltStack, Fabric, Terraform …<br />
+            Process of managing and provisioning servers through definition files rather than configuration tools.<br />
+            Promoted for cloud computing, often confused with Infrastructure as a Service (IaaS)<br />
+            This brought forth 3 improvements: Cost reduction, faster execution speed, and restricted risk (remove errors and security violations).<br />
```
Comment (martinheidegger) [inline 20180224_kubernetes/index.html:86]: Succinct version:

3 improvements:
- Less Cost
- Faster
- Less risk

**Review by martinheidegger — COMMENTED** [review]
In the great picture your presentation is very common to other orchestration presentations in that it is very hard to follow what you are talking about.

I proposed alternative language on the slides in the hope that you can draw a picture that people understand: Less words to read = less words to understand.

During the presentation we have a white-board the projector is projecting to. Maybe you should use this to write down all the new vocabulary.

Also it might be worth noting that people usually want to know why its important that we call a Pod a pod and why we should know why this can not be simpler.

In the summary _(which is good to have 👍)_ It combines all those concepts but it doesn't quite use the language that you used throughout the presentation. It would be good if the summary at the end only contains the things you talked about.

Another thought: People don't know why they should/have-to bother with all this docker stuff... It would be good to bring a workflow example - that is relate-able - as a basis for your work. Something like "Imagine you have a business hosting websites, you could update all those by hand or you could use kubernetes. Kubernetes even makes sure to restart your things if they ... god-forbid ... break down"

File: 20180224_kubernetes/index.html (hunk 27 above)
Comment (AdrienLemaire) [reply 20180224_kubernetes/index.html]: Yeah, I'm not done with the presentation yet ... X_X Haven't even mentioned Deployments.

**Comment by AdrienLemaire** [comment]
Thanks for the review @martinheidegger , will try to do better in future presentations :)
